iceoryx2-tunnels/zenoh/src/discovery/zenoh.rs

```diff
@@ -59,36 +69,49 @@ impl<ServiceType: iceoryx2::service::Service> ZenohDiscovery<'_, ServiceType> {
 impl<ServiceType: iceoryx2::service::Service> Discovery<ServiceType>
     for ZenohDiscovery<'_, ServiceType>
 {
-    fn discover<OnDiscovered: FnMut(&IceoryxServiceConfig)>(
+    fn discover<OnDiscovered: FnMut(&ServiceConfig) -> Result<(), DiscoveryError>>(
         &mut self,
         on_discovered: &mut OnDiscovered,
     ) -> Result<(), DiscoveryError> {
         // Drain all replies from previous query
-        for z_reply in self.z_query.drain() {
-            match z_reply.result() {
-                Ok(z_sample) => {
-                    match serde_json::from_slice::<IceoryxServiceConfig>(
-                        &z_sample.payload().to_bytes(),
-                    ) {
-                        Ok(iox_service_details) => {
-                            on_discovered(&iox_service_details);
+        for reply in self.replies.drain() {
+            match reply.result() {
+                Ok(sample) => {
+                    match serde_json::from_slice::<ServiceConfig>(&sample.payload().to_bytes()) {
+                        Ok(service_details) => {
+                            fail!(
+                                from &self,
+                                when on_discovered(&service_details),
+                                "failed to process remote service discovered via Zenoh"
+                            )
+                        }
+                        Err(e) => {
+                            warn!(
+                                "skipping discovered service config, unable to deserialize: {}",
+                                e
+                            );
                         }
-                        Err(_e) => todo!(),
                     }
                 }
-                Err(_e) => { /* Ignore and process other requests */ }
+                Err(e) => fail!(
+                    from "discovery_via_subscriber()",
+                    when Err(e),
```

Comment:
You can remove this. But it would be nice to print out the error value in the error message.
```
"errorneous reply received from zenoh discovery query ({e:?})"
```